metadata-ingestion/src/datahub/sql_parsing/sqlglot_lineage.py

```diff
@@ -949,6 +1005,50 @@ def _list_joins(
                 )
             )
 
+        # Handle LATERAL constructs
+        for lateral in scope.expression.find_all(sqlglot.exp.Lateral):
+            try:
+                # Get tables from non-lateral FROM clauses
+                qualified_left: OrderedSet[_TableName] = OrderedSet()
+                for from_clause in scope.find_all(sqlglot.exp.From):
+                    if not isinstance(from_clause.this, sqlglot.exp.Lateral):
+                        qualified_left.update(
+                            t.qualified(
+                                dialect=dialect,
+                                default_db=default_db,
+                                default_schema=default_schema,
+                            )
```

Comment:
by this point in the processing, all table refs should already be qualified. this logic seems extremly strange, although I don't fully understand what changed in sqlglot recently